nemo/collections/nlp/data/language_modeling/megatron/data_samplers.py

```diff
@@ -149,9 +150,13 @@ def __init__(
             pad_samples_to_global_batch_size == False
         ), "`MegatronPretrainingRandomSampler` does not support sample padding"
         self.last_batch_size = self.total_samples % self.micro_batch_times_data_parallel_size
+        self.seed = seed
 
     def __len__(self):
-        num_available_samples: int = self.total_samples
```

Comment:
It seems like previously `__len__()` was ignoring `consumed_samples`. This seems like a bug to me (that this PR would fix) but it'd be good to get the confirmation from whoever wrote that class to make sure it's not intended.

Reply:
Will find out and check